accessibility/common-remediations.md

```diff
@@ -0,0 +1,197 @@
+# Common remediations
+
+This page describes some common accessibility issues, and gives examples and suggestions for repairs. 
+
+
+## Keyboard navigation
+
+### Missing or inadequate focus styles
+
+Not everyone uses a mouse. Some users navigate through a page with their keyboard by pressing the <kbd>tab</kbd> key to move from one focusable element to another. If focus styles are removed or insufficient, these users have no way of knowing which element they are about to activate. 
+
+All elements that receive focus MUST have visible focus styles. 
+
+Browser default focus styles are likely insufficient as they can be difficult to see in some designs. We prefer focus styles to be highly visible. 
+
+
+We do this:
+```css
+a:focus {
+   outline: 3px solid #fece3e;
+}
+
+a:focus, button:focus {
+   background-color: #fc0;
+   outline: 2px solid #fc0;
+}
+```
+
+We _don't_ do this:
+```css
+a:focus {
+   outline: 0;
+}
+```
+
+
+### Interactive components not accessible by keyboard
+
+#### Unsemantic elements used for interactive components
+
+Built-in interactive HTML elements like anchors, text fields, buttons, and select lists (among others) are handled automatically by the browser. Users can interact with them with the mouse, the keyboard, by voice, or any other kind of technology that they like. 
+
+Elements with no semantics (divs and spans) don’t have this automatic handling, and a simple `onclick` handler doesn’t provide it. These elements generally can’t be interacted with by users of Assistive Technology unless you write large amounts of JavaScript to make it possible. 
+
+Where a built-in interactive element is available, use it. 
+
+We do this:
+```html
+<button>Toggle UI</button>
+<a href=”/”>Visit this location</a>
+```
+We _don’t_ do this:
+```html
+<div onclick=”doFunc();”>Toggle UI</div>
+<span>Visit this location</a>
+```
+
+
+#### Links with no href
+
+All anchor links (the `a` element) require an `href` attribute in order to be functional as links. Omitting the `href` attribute prevents keyboard users and voice input users from being able to use the link at all. 
+
+We do this:
+```html
+<a href=”#location”>Visit this location</a>
+```
+We _don’t_ do this:
+```html
+<a onclick=”myFunc();”>Visit this location</a>
+```
+
+
+
+####  Anchor elements with the [tabindex=-1] attribute are not usable by keyboard users.
```

Comment:
```suggestion
####  Inappropriate negative tabindex

Interactive elements with a negative tabindex attribute (`tabindex=-1`) are not usable by keyboard users. 

```

I've generalised this a little because it could be applied incorrectly to a button as well as a link